libraries/AP_RCTelemetry/AP_VideoTX.h

```diff
@@ -0,0 +1,131 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Param/AP_Param.h>
+
+#define VTX_MAX_CHANNELS 8
+
+class AP_VideoTX {
+public:
+    AP_VideoTX();
+    ~AP_VideoTX();
+
+    /* Do not allow copies */
+    AP_VideoTX(const AP_VideoTX &other) = delete;
+    AP_VideoTX &operator=(const AP_VideoTX&) = delete;
+
+    // init - perform required initialisation
+    bool init();
+
+    // run any required updates
+    void update();
+
+    static AP_VideoTX *get_singleton(void) {
+        return singleton;
+    }
+    static const struct AP_Param::GroupInfo var_info[];
+
+    enum class VideoOptions {
+        VTX_PITMODE = 1 << 0
+    };
+
+    enum VideoBand {
+        VTX_BAND_A,
+        VTX_BAND_B,
+        VTX_BAND_E,
+        VTX_BAND_FATSHARK,
+        VTX_BAND_RACEBAND,
+        VTX_BAND_LOW_RACEBAND,
+        MAX_BANDS
+    };
+
+    static const uint16_t VIDEO_CHANNELS[MAX_BANDS][VTX_MAX_CHANNELS];
+
+    static uint16_t get_frequency_mhz(uint8_t band, uint8_t channel) { return VIDEO_CHANNELS[band][channel]; }
+    static bool get_band_and_channel(uint16_t freq, VideoBand& band, uint8_t& channel);
+
+    void set_frequency_mhz(uint16_t freq) { _frequency_mhz.set_and_save(freq); }
+    uint16_t get_frequency_mhz() const { return _frequency_mhz; }
+    // get / set power level
+    void set_power_mw(uint16_t power) { _current_power = power; }
+    void set_power_dbm(uint8_t power);
+    void set_configured_power_mw(uint16_t power, bool force = true);
+    uint16_t get_configured_power_mw() const { return _power_mw; }
+    uint16_t get_power_mw() const { return _current_power; }
+    uint8_t get_configured_power_dbm() const;
+    bool update_power() const { return _power_mw != _current_power && _power_mw.configured_in_storage(); }
```

Comment:
best to avoid these configured_in_storage() calls at runtime, as discussed

Reply:
Fixed